### ISSUE TITLE
Discover Algorithm tends to play the same Artist in a row

### ISSUE THREAD

**Issue by vbinetruy** [MASASFM/masas#51]
Although this does not happen all the time, Discover tends to play the same Artist (but different songs of that Artist) one after another before playing songs from other Artist. You can see what I mean in the screenshot attached below (this Artist has the same cover photo again and again, but the songs are all different). 

A solution that we could use for this issue is to add a variable in the algorithm that prevents the system from playing the same Artist's music in a row.

To prevent getting a list of music that would be "Artist 1 song 1" "Artist 1 song 2"
"Artist 1 song 3" "Artist 1 song 4" "Artist 1 song 5"

<img width="1280" alt="screen shot 2016-12-07 at 1 12 11 pm" src="https://cloud.githubusercontent.com/assets/19936607/20972538/941d0cb2-bcae-11e6-8d53-63b87adf6e95.png">

**Comment by jpic** [comment]
If we implemented what you suggested, then we'd always be playing the first song of each serie of upload per artist.

**Comment by vbinetruy** [comment]
So how can we fix this problem?



<img width="1280" alt="screen shot 2016-12-07 at 7 31 57 pm" src="https://cloud.githubusercontent.com/assets/19936607/20973944/f657d8c6-bcb3-11e6-974c-56ba1a1dec8e.png">
![Uploading Screen Shot 2016-12-07 at 7.32.09 PM.png…]()

**Comment by vbinetruy** [comment]
<img width="1280" alt="screen shot 2016-12-07 at 7 32 17 pm" src="https://cloud.githubusercontent.com/assets/19936607/20973999/37d2f524-bcb4-11e6-9a9b-70d63250c153.png">

**Comment by vbinetruy** [comment]
<img width="1280" alt="screen shot 2016-12-07 at 7 35 49 pm" src="https://cloud.githubusercontent.com/assets/19936607/20974054/650b7048-bcb4-11e6-87ad-95dcd6f95853.png">

**Comment by vbinetruy** [comment]
<img width="1280" alt="screen shot 2016-12-07 at 7 38 58 pm" src="https://cloud.githubusercontent.com/assets/19936607/20974196/d1de7ddc-bcb4-11e6-907b-4f2db9db4140.png">

**Comment by tbinetruy** [comment]
James and i fucked up yesterday evening. The bug is because we ALWAYS play songs in their upload order. meaning that no matter who you are and what songs you've already listened too, whenever you connect to MASAS' discover you will head songs in their upload order, with most recent first. And if an artist upload 20 songs, then they will always be played the one after the other on discover. (hence the bug on your first screenshot). 

**THIS ISSUE IS THEREFORE NOT CLOSED**

It's terrible design, we need to redo it. I believe there are even cases where a given song might never be heard which is unacceptable (it's the opposite of what masas sells).

The reason we did it this was is to open up popular ASAP. But agreed that it can't be live as it is.

As a side note, james and I fucked up big time by deploying to prod branch instead of dev2 branch for testing.

I think for now the best is to revert to the last commit before we pushed to prod yesterday.

As for the near future, I'm going to make tests for both popular and discover algorithms.
